source/develop/release-management/features/virt/vm-screenshot.md

```diff
@@ -0,0 +1,83 @@
+---
+title: VM Screenshot
+category: feature
+authors: sabusale
+---
+
+# VM Screenshot
+
+## Summary
+
+Allow users to take screenshot of the VM
+
+## Owner
+
+*   Name: Saif Abu Saleh (sabusale)
+*   Email: sabusale@redhat.com
+
+## Description
+
+User can take a screenshot of running VM
+
+### UI
+
+Not supported
+
+### Backend
+
+Backend will call VDSM to capture screenshot and will receive the `ppm`[[1]] file content
+and then `ppm` file content will be returned
+
+### VDSM
+
+VDSM will call Libvirt API[[2]] that capture the screenshot and saves it on given path on the host file system
+and then will return the `ppm` file content and remove the image on the host file system
+
+Saving the image:
+*   path will be temporary directory path
+*   type is `ppm` which is the only provided option by QEMU [[3]]
+
+
+Other options: 
+*   Save all images to known `static` path and keep them, save with special file names, e.g: 
+    `screenshot_{vm}_{timestamp}`
+
+### REST API
+
+`/vms/{vm:id}/screenshot`
+
+```
+      POST /vms/1234/screenshot
+      Accept: application/xml
+```
+
+```
+<screenshot>
+<data>iVBORw0K..</data>
+<media_type>image/x-portable-pixmap</media_type>
+</screenshot>
+```
+
+### Image format issues
+
+`ppm` image is not a standard image format, and users may have issues to see the image, and it is not straight forward
+to display it in HTML[[4]], unlike e.x `png base64`
+
+other option is to convert the `ppm` content to other standard format like `png`
+ways to convert it:
+*   Run command line command to convert it using ImageMagick
+*   Use java API to convert it using ImageMagick , e.x: https://stackoverflow.com/a/2291493/5825468
+and then encode with `base64`, similar to icons
```

Comment:
I think that as it is mainly intended for debugging, users can install GIMP or any other tool that opens ppm
but yeah, it would be nice to design this operation to be extensible so if, e.g., a UI client would like to present it, it can ask for receiving it in a given format (like png) and the engine will convert it